crates/cdk-integration-tests/tests/fake_wallet.rs

```diff
@@ -377,6 +378,87 @@ async fn test_fake_melt_change_in_quote() -> Result<()> {
     Ok(())
 }
 
+#[tokio::test(flavor = "multi_thread", worker_threads = 1)]
+async fn test_fake_mint_with_witness() -> Result<()> {
+    let wallet = Wallet::new(
+        MINT_URL,
+        CurrencyUnit::Sat,
+        Arc::new(WalletMemoryDatabase::default()),
+        &Mnemonic::generate(12)?.to_seed_normalized(""),
+        None,
+    )?;
+    let secret = SecretKey::generate();
+    let mint_quote = wallet
+        .mint_quote(100.into(), None, Some(secret.public_key()))
+        .await?;
+
+    wait_for_mint_to_be_paid(&wallet, &mint_quote.id).await?;
+
+    let mint_amount = wallet
+        .mint(&mint_quote.id, SplitTarget::default(), None, Some(secret))
+        .await?;
```

Comment:
Would it be good to generate `secret` inside `wallet.mint_quote` and also put it into the database there, so when you call `wallet.mint()` a few lines further, you don't need to provide `secret.public_key()` but could fetch it from the db again inside `wallet.mint()`?

Reply:
Yeah I was thinking about this I initially did it this way to avoid storing the secret in the database but I don't think there is much benefit to that. Storing it in the cdk database makes it much easier for the implementer as they do not have to store or manage secrets so I will go this route as you suggest.